test/test_resonator_spectroscopy.py

```diff
@@ -0,0 +1,94 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Spectroscopy tests for resonator spectroscop experiment."""
+
+from test.base import QiskitExperimentsTestCase
+from typing import Tuple
+import numpy as np
+
+from qiskit import QuantumCircuit
+
+from qiskit_experiments.library import ResonatorSpectroscopy
+from qiskit_experiments.test.mock_iq_backend import MockIQBackend
+
+
+class ResonatorSpectroscopyBackend(MockIQBackend):
+    """A simple and primitive backend to test spectroscopy experiments."""
+
+    def __init__(
+        self,
+        line_width: float = 2e6,
+        freq_offset: float = 0.0,
+        iq_cluster_centers: Tuple[float, float, float, float] = (-1.0, 0.0, 0.0, 0.0),
+        iq_cluster_width: float = 0.2,
+    ):
+        """Initialize the spectroscopy backend."""
+
+        super().__init__(iq_cluster_centers, iq_cluster_width)
+
+        self._linewidth = line_width
+        self._freq_offset = freq_offset
+
+        super().__init__(iq_cluster_centers, iq_cluster_width)
+
+    def _compute_probability(self, circuit: QuantumCircuit) -> float:
+        """Returns the probability based on the frequency."""
+        freq_shift = next(iter(circuit.calibrations["measure"].values())).blocks[0].frequency
+        delta_freq = freq_shift - self._freq_offset
+
+        return np.exp(-(delta_freq ** 2) / (2 * self._linewidth ** 2))
```

Comment:
Lorentzian would be more physically accurate so I changed it here: b5bce85. however I needed to remove the fit quality `good` assert and reduce the precision. This is because `ResonanceAnalysis` uses a Gaussian fit. I opened an issue here: https://github.com/Qiskit/qiskit-experiments/issues/617